projects/igniteui-angular/src/lib/toast/toast.component.ts

```diff
@@ -277,17 +297,42 @@ export class IgxToastComponent implements IToggleView, OnInit, OnDestroy {
      */
     public show(message?: string): void {
         clearInterval(this.timeoutId);
-        if (message !== undefined) { this.toastMessage = message; }
+
+        const overlaySettings: OverlaySettings = {
+            positionStrategy: new GlobalPositionStrategy({
+                horizontalDirection: HorizontalAlignment.Center,
+                verticalDirection:
+                    this.position === 'bottom'
+                        ? VerticalAlignment.Bottom
+                        : this.position === 'middle'
+                        ? VerticalAlignment.Middle
+                        : VerticalAlignment.Top,
+            }),
+            closeOnEscape: false,
+            closeOnOutsideClick: false,
+            modal: false,
+            outlet: this.outlet
+        };
+
+        if (message !== undefined) {
+            this.toastMessage = message;
+        }
+
+        this.overlayId = this.overlayService.attach(
+            this.elementRef,
+            overlaySettings
+        );
+
         this.onShowing.emit(this);
-        this.isVisible = true;
-        this.animationState = 'visible';
+        this._isVisible = true;
 
         if (this.autoHide) {
             this.timeoutId = window.setTimeout(() => {
                 this.hide();
             }, this.displayTime);
         }
 
+        this.overlayService.show(this.overlayId);
         this.onShown.emit(this);
```

Comment:
this will fire too early before toast is actually shown. You should subscribe to `IgxOverlay`'s `onOpened` event and emit `onShown` there. IMO it will be much more easier if you try to use `IgxToggleDirective` instead of `IgxOverlayService`.

Reply:
This should be handled better now since the event is fired on the `onOpened` event of the toggle directive.